pkg/lifecycle/kustomize/pre_kustomize.go

```diff
@@ -95,3 +109,87 @@ func (l *Kustomizer) maybeSplitListYaml(ctx context.Context, path string) error
 
 	return nil
 }
+
+func (l *Kustomizer) initialKustomizeRun(ctx context.Context, step api.Kustomize) error {
+	if err := l.writeBase(step); err != nil {
+		return errors.Wrap(err, "write base kustomization")
+	}
+
+	built, err := l.kustomizeBuild(step.Base)
+	if err != nil {
+		return errors.Wrap(err, "build overlay")
+	}
+
+	if err := l.writePostKustomizeFiles(step, built); err != nil {
+		return errors.Wrap(err, "write initial kustomized yaml")
+	}
+
+	if err := l.replaceOriginal(step, built); err != nil {
+		return errors.Wrap(err, "replace original yaml")
+	}
+
+	return nil
+}
+
+func (l *Kustomizer) replaceOriginal(step api.Kustomize, built []postKustomizeFile) error {
```

Comment:
I would *love* a test for this
It looks like it should work fine with our mocked FS for testing, right?

Reply:
Which exists already, and I completely missed. Right.